tests/system/src/test/java/test/javafx/embed/swing/SwingNodeContentMemoryLeakTest.java

```diff
@@ -0,0 +1,137 @@
+import java.util.Collection;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CountDownLatch;
+
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import javafx.application.Application;
+import javafx.embed.swing.SwingNode;
+import javafx.scene.Scene;
+import javafx.scene.layout.Pane;
+import javafx.stage.Stage;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.sun.javafx.PlatformUtil;
+
+import test.util.Util;
+
+public class SwingNodeContentMemoryLeakTest {
+
+    static CountDownLatch launchLatch = new CountDownLatch(1);
+
+    //Keep week references to all panels that we've ever generated to see if any
+    //of them get collected.
+    private Collection<WeakReference<JPanel>> panels = new CopyOnWriteArrayList<>();
+    private int count = 0;
+    private int fail = 0;
+    private SwingNode node;
+
+    @BeforeClass
+    public static void setupOnce() {
+        Util.launch(launchLatch, 50, MyApp.class);
+    }
+
+    @AfterClass
+    public static void teardownOnce() {
+        Util.shutdown();
+    }
+
+    @Test
+    public void testSwingNodeContentMemoryLeak() throws InterruptedException,
+                                                        InvocationTargetException {
+        Util.runAndWait(() -> {
+            node = new SwingNode();
+            Pane root = new Pane();
+            root.getChildren().add(node);
+
+            Stage stage = new Stage();
+            Scene scene = new Scene(root, 150, 100);
+            stage.setScene(scene);
+        });
+
+        //Kick off a thread that repeatedly creates new JPanels and resets the swing node's content
+        new Thread(() -> {
+            while(count < 50) {
+                //Lets throw in a little sleep so we can read the output
+                try {
+                    Thread.sleep(100);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+
+                SwingUtilities.invokeLater(() -> {
+                    JPanel panel = new JPanel();
+                    panels.add(new WeakReference<>(panel));
+                    node.setContent(panel);
+                });
+
+                long panelCount = panels.stream().filter(ref ->
+                                                     ref.get() != null).count();
+                // Sometimes panel count can shoot upto more than 3 once or twice
+                // due to gc not being guranteed so this check prevents false failure
+                // Without fix, the panel count will increase continuosly so it will
+                // always be more 3 after 1-2 iterations
+                if (panelCount > 3) {
+                    fail++;
+                }
+                System.out.println("iteration " + count + " Panels in memory: "
+                                               + panelCount + " fail " + fail);
+                assertFalse(fail > 2);
```

Comment:
I actually seen the  number to shoot up to 38 (albeit with Thread.sleep(1)).  I wonder if there is a better way to detect failure?  May be the fact that the 'panelCount` is ever increasing is the sign of failure, whereas if at least one time it's dropping we have succeeded.
Or perhaps look at the count when the test complete and fail if panelCount < (attempts / 2) or something

Reply:
I am not sure I understand the nuances of JMemoryBuddy. I tried using `JMemoryBuddy.assertCollectable` but it fails with 
```
Exception in thread "Thread-7" java.lang.AssertionError: Content of WeakReference was not collected. content: javax.swing.JPanel[,0,0,0x0,layout=java.awt.FlowLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=9,maximumSize=,minimumSize=,preferredSize=]
	at test.util.memory.JMemoryBuddy.assertCollectable(JMemoryBuddy.java:91)
```

maybe because the test doesn't cause SwingNode content count to 0.

But I have modified the test to have better way of detecting failure as suggested and check for total count at end which normally increases without fix..I hope that should be good enough to ensure it fails without the fix and pass with it..

Reply:
Any more comments on this test fix after updation @andy-goryachev-oracle ?